eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -225,13 +227,18 @@ class EclairImpl(appKit: Kit) extends Eclair {
     * @param request
     * @return
     */
-  def sendToChannel(channelIdentifier: String, request: Any)(implicit timeout: Timeout): Future[Any] =
-    for {
-      fwdReq <- Future(Register.ForwardShortId(ShortChannelId(channelIdentifier), request))
-        .recoverWith { case _ => Future(Register.Forward(ByteVector32.fromValidHex(channelIdentifier), request)) }
-        .recoverWith { case _ => Future.failed(new RuntimeException(s"invalid channel identifier '$channelIdentifier'")) }
-      res <- appKit.register ? fwdReq
-    } yield res
+  def sendToChannel(channelIdentifier: String, request: Any)(implicit timeout: Timeout): Future[Any] = {
+    (Try(ForwardShortId(ShortChannelId(channelIdentifier), request)) match {
+      case Success(value) => Left(value.shortChannelId)
+      case Failure(_) => Try(Forward(ByteVector32.fromValidHex(channelIdentifier), request)) match {
+        case Success(v) => Right(v.channelId)
```

Comment:
Why not call `appKit.register ? Forward(channelId, request)` here ?

Reply:
How about 125cbbf?